src/python-common/ceph/deployment/service_spec.py

```diff
@@ -227,13 +226,13 @@ def from_string(cls, arg):
         PlacementSpec(count=3, label='mon')
 
         fnmatch is also supported:
-        >>> PlacementSpec.from_string('host_pattern:data[1-3]')
+        >>> PlacementSpec.from_string('data[1-3]')
         PlacementSpec(host_pattern='data[1-3]')
 
         >>> PlacementSpec.from_string(None)
         PlacementSpec()
         """
-        if arg is None:
+        if arg is None or not arg:
```

Comment:
``or arg == ''``?  otherwise ``0`` won't produce count=0?

Reply:
should work!
```python
In [1]: not None
Out[1]: True

In [2]: not ''
Out[2]: True

In [3]: not '0'
Out[3]: False
```

Reply:
ah, str, got it.